### PR TITLE
[master-2.x] Add try-catch when receive a new tx

### PR DESCRIPTION
**Pull request by Tommo-L**
close https://github.com/neo-project/neo/issues/2065

### CODE REVIEW

**Review by shargon — APPROVED** [review]

**Comment by superboyiii** [comment]
Works well on mainnet now.

**Comment by superboyiii** [comment]
Works well on mainnet. It's compatible.

**Comment by Tommo-L** [comment]
Merge?

File: neo/Ledger/Blockchain.cs [hunk 1]
```diff
@@ -393,20 +393,26 @@ private void OnNewHeaders(Header[] headers)
...
+                    return RelayResultReason.Invalid;
+                if (!Plugin.CheckPolicy(transaction))
+                    return RelayResultReason.PolicyFail;
 
+                if (!MemPool.TryAdd(transaction.Hash, transaction))
+                    return RelayResultReason.OutOfMemory;
+            }
+            catch
+            {
+                return RelayResultReason.Invalid;
```
Comment (shargon) [inline neo/Ledger/Blockchain.cs:414]: Create `Error`?
```suggestion
                return RelayResultReason.Error;
```

Comment (Tommo-L) [reply neo/Ledger/Blockchain.cs]: Done

**Review by shargon — APPROVED** [review]

**Pull request merged**